src/pages/dataSetsWip/form/dataEntryForm/customForm/CustomFormEdit.tsx

```diff
@@ -0,0 +1,285 @@
+    StandardFormSectionDescription,
+    StandardFormSectionTitle,
+} from '../../../../../components'
+import { generateDhis2Id } from '../../../../../lib/index'
+import styles from './CustomFormContents.module.css'
+import { CustomFormElementsSelector } from './CustomFormElementsSelector'
+
+export type CustomFormProps = {
+    closeCustomFormEdit?: () => void
+}
+
+const SubsectionSpacer = ({ children }: { children: React.ReactNode }) => (
+    <div className={styles.formSectionSpacing}>{children}</div>
+)
+
+const getElementText = ({
+    id,
+    name,
+    type,
+    disabled,
+}: {
+    id: string
+    name?: string
+    type: string
+    disabled?: boolean
+}): string => {
+    if (type === 'dataElement') {
+        const [de, coc] = id.split('.')
+        return `<p><input ${
+            disabled ? 'disabled="disabled"' : ''
+        } id="${de}-${coc}-val" name="entryfield" title="${name}" value="[ ${name} ]" /></p>`
+    }
+    if (type === 'total') {
+        return `<p><input dataelementid="${id}" ${
+            disabled ? 'disabled="disabled"' : ''
+        } id="total${id}" name="total" readonly="readonly" title="${name}" value="[ ${name} ]" /></p>`
+    }
+    if (type === 'indicator') {
+        return `<p><input ${
+            disabled ? 'disabled="disabled"' : ''
+        } id="indicator${id}" indicatorid="${id}" name="indicator" readonly="readonly" title="${name}" value="[ ${name} ]" /></p>`
+    }
+    if (type === 'flag') {
+        return `<p><img src="../dhis-web-commons/flags/${id}.png" /></p>`
+    }
+    return ''
+}
+
+type CustomFormDataPayload = {
+    id: string
+    htmlCode: string
+}
+// when there is a data set id, we can post to the form endpoint
+const useUpdateForm = ({
+    onSuccess,
+    onError,
+}: {
+    onSuccess: (data: CustomFormDataPayload) => void
+    onError: (e: Error) => void
+}) => {
+    const id = useParams().id
+    const dataEngine = useDataEngine()
+
+    const update = useCallback(
+        async (data: CustomFormDataPayload) => {
+            try {
+                const response = await dataEngine.mutate(
+                    {
+                        resource: `dataSets/${id}/form`,
+                        type: 'create',
+                        data: data,
+                    },
+                    {
+                        onComplete: () => {
+                            // the response from this post is empty, so we use the data we passed if it was successful
+                            onSuccess(data)
+                        },
+                        onError,
+                    }
+                )
+                return { data: response }
+            } catch (error) {
+                console.error(error)
+            }
+        },
+        [dataEngine, id, onSuccess, onError]
+    )
+    return update
+}
+
+export const CustomFormEdit = ({ closeCustomFormEdit }: CustomFormProps) => {
+    const textAreaRef = useRef<HTMLTextAreaElement>(null)
```

Comment:
I wanted an uncontrolled component, so that the state is not constantly updating, hence this and the use of the native <textarea> element